tests: kernel: timer: timer_behavior: Tweak expected std deviation

nordic-krch opened the pull request:
If frequency of the system clock is lower then deviation may exceed default value (10us). Instead of adjusting the default value, test is rounding up expected standard deviation to a single clock cycle.

Fixes #71863.

nordic-krch committed:
tests: kernel: timer: timer_behavior: Tweak expected std deviation

If frequency of the system clock is lower then deviation may exceed
default value (10us). Instead of adjusting the default value, test is
rounding up expected standard deviation to a single clock cycle.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

teburd commented:
I like the change, agree with @golowanow the calculation should be done once before the TC_PRINT and zaasert usage

andyross reviewed (dismissed):
Seems reasonable.  FWIW, an alternate technique would be to do the tracking and math in cycle space and not microseconds and just clamp the minimum to "1", which IMHO would be clearer but require more rework.

nordic-krch committed:
tests: kernel: timer: behavior: pytest: Adjust max stddev

Like in C test. If MAX STDDEV is lower than single clock cycle then
set it to a single clock cycle.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

golowanow commented on tests/kernel/timer/timer_behavior/src/jitter_drift.c:
```diff
@@ -283,7 +285,7 @@ static void do_test_using(void (*sample_collection_fn)(void), const char *mechan
 		 ", \"CONFIG_TIMER_TEST_PERIOD\":%d"
 		 ", \"CONFIG_TIMER_TEST_SAMPLES\":%d"
-		 ", \"CONFIG_TIMER_TEST_MAX_STDDEV\":%d"
+		 ", \"MAX STD DEV\":%d"
```
same recording output change should be in pytest/test_timer.py as well

golowanow replied:
why the record name is different now ?

aescolar commented:
@andyross @peter-mitsis please review

golowanow commented on tests/kernel/timer/timer_behavior/pytest/test_timer.py:
```diff
@@ -62,13 +65,13 @@ def do_analysis(test, stats, stats_count, config, sys_clock_hw_cycles_per_sec):
                 f', "min_bound_us":{min_bound * 1_000_000:.6f}'
                 f', "max_bound_us":{max_bound * 1_000_000:.6f}'
                 f', "expected_period_cycles":{expected_period:.0f}'
+                f', "max_stddev":{max_stddev:.6f}'
```
```suggestion
                f', "MAX_STD_DEV":{max_stddev:.6f}'
```
record field name should be the same if it is the same value with jitter_drift.c

nordic-krch replied:
done

golowanow reviewed (approved):
LGTM

andyross commented on tests/kernel/timer/timer_behavior/pytest/test_timer.py:
```diff
@@ -37,7 +37,10 @@ def do_analysis(test, stats, stats_count, config, sys_clock_hw_cycles_per_sec):
     max_bound = (test_period + period_max_drift * test_period +
                  expected_period_drift) / 1_000_000
 
+    cyc_us = 1000000 / sys_clock_hw_cycles_per_sec
```
Naming whine: many platforms have cycle rates > 1MHz, making this value a zero.   Giving it a name that looks and sounds like "cycles per microsecond" is likely to end up in a divide-by-zero as the code evolves.  It's used as a clamp, so e.g. "`min_cyc = ...; /* Minimum possible cycles that can be measured */`' might be better?

andyross reviewed (approved):
Still looks fine to me.  One nitpick.
